uml-diagrams/CommandClass.puml

```diff
@@ -0,0 +1,89 @@
+@startuml
+skinparam groupInheritance 2
+abstract class Command {
+    +createJsonObject(): JsonObject
+}
+
+abstract class CheckInformationCommand extends Command {
+    +execute(userInput: String): void {abstract}
+}
+
+class FilterCoursesCommand extends CheckInformationCommand {
+    -logger: Logger {static}
+    +execute(userInput: String): void
+    +getNusCourseCode(userInput: String): String
+    +displayMappableCourses(jsonObject: JsonObject, courseToFind: String): void
+    +isCourseFound(courseToFind: String, universityName: String, numberOfCourses: int,
+         courses: JsonArray, isCourseFound: boolean): boolean
+    +printMappableCourse(universityName: String, course: JsonObject): void
+}
+
```

Comment:
The diagram is very detailed which is great I thought we just needed the brief overview oops.

Thought it would look something like this
![image](https://github.com/user-attachments/assets/a7a3bf43-0196-42d2-a746-5200e50ca60f)

But regardless let's just put this in the DG :)

Reply:
LGTM but please fix merge conflicts. @chewycj

Reply:
I have fixed the merge conflicts! I was familiar with the conflict as it was part of my code. 
This raised a compilation error due to different `ListPersonalTrackerCommand` constructor in parser. I have highlighted the line of the code to change.